File: examples/stack-samples/issue-144-java/src/main/java/com/example/userapi/service/UserService.java
package com.example.userapi.service;

import java.util.List;
import java.util.Optional;
import com.example.userapi.model.User;
import com.example.userapi.repository.UserRepository;

public class UserService {
    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public Optional<User> getUserById(Long id) {
        return userRepository.findById(id);
    }

    public User createUser(String name, String email) {
        User user = new User(null, name, email);
        return userRepository.save(user);
    }

    public void deleteUser(Long id) {
        userRepository.deleteById(id);
    }

    private void validateUser(User user) {
        if (user.getName() == null || user.getName().isEmpty()) {
            throw new IllegalArgumentException("Name is required");
        }
    }
}

File: examples/stack-samples/issue-145-php/app/Services/UserService.php
<?php

namespace App\Services;

use App\Models\User;

class UserService
{
    private array $users = [];

    public function getAllUsers(): array
    {
        return $this->users;
    }

    public function findUser(int $id): ?User
    {
        return $this->users[$id] ?? null;
    }

    public function createUser(string $name, string $email): User
    {
        $user = new User(count($this->users) + 1, $name, $email);
        $this->users[] = $user;
        return $user;
    }

    public function deleteUser(int $id): bool
    {
        if (isset($this->users[$id])) {
            unset($this->users[$id]);
            return true;
        }
        return false;
    }

    private function validateUser(User $user): void
    {
        if (empty($user->getName())) {
            throw new \InvalidArgumentException("Name required");
        }
    }
}
